frontend/packages/console-dynamic-plugin-sdk/README.md

```diff
@@ -447,3 +447,94 @@ The list of shared modules planned for deprecation:
 [console-doc-api]: ./docs/api.md
 [console-doc-feature-page]: https://github.com/openshift/enhancements/blob/master/enhancements/console/dynamic-plugins.md
 [console-pf-upgrade-notes]: ./upgrade-PatternFly.md
+
+## i18n traslations for messages
+
+The following demonstrates how to translate messages in the console plugin using the [i18next](https://www.i18next.com/) and [react-i18next](https://react.i18next.com/) libraries. Also included are the instructions for uploading/downloading strings to/from the Phrase Translation Memory System (TMS).
+
+### Step 1: Mark strings with `t` function for traslation
+
+You can use the `useTranslation` hook within the component, or the `i18next` function outside the component, along with the namespace. The i18n namespace must match the name of the `ConsolePlugin` resource with the `plugin__` prefix to avoid naming conflicts. For example, a console plugin named console-plugin-hello-world would use the `plugin__console-plugin-hello-world` namespace. See the following for examples:
+
+```tsx
+import i18next from 'i18next';
+import { useTranslation } from 'react-i18next';
+
+const helloWorldMessages = {
+  Foo: {
+    title: i18next.t('plugin__console-plugin-hello-world~Foo'),
+    message: i18next.t(
+      'plugin__console-plugin-hello-world~This is foo description...',
+    ),
+  },
+    Bar: {
+    title: i18next.t('plugin__console-plugin-hello-world~Bar'),
+    message: i18next.t(
+      'plugin__console-plugin-hello-world~This is bar description...',
+    ),
+  },
+  }
+
+const Header: React.FC = () => {
+  const { t } = useTranslation('plugin__console-plugin-hello-world');
+  return <h1>{t('Hello, World!')}</h1>;
+};
+```
+
+For labels in `console-extensions.json`, you can use the format `%plugin__console-plugin-hello-world~My Label%`. Console will replace the value with the string for the current language from the `plugin__console-plugin-hello-world` namespace. For example:
+
+```json
+  {
+    "type": "console.navigation/section",
+    "properties": {
+      "id": "admin-demo-section",
+      "perspective": "admin",
+      "name": "%plugin__console-hello-world~Hello World%"
```

Comment:
We should use a single plugin name for these examples. I see that we have two here - `console-hello-world` and `console-plugin-hello-world` used in the `Header` function.
Also lets rather use a real example, like the `console-crontab-plugin`.